### PR TITLE
Added a PR template

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details

## Description

Added a pull request template. Doesn't actually work until it is in the default branch of the repository (develop). I manually added it into this description to illustrate what it looks like.

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ ] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [ ] My changes require updates and/or additions to the unit tests:
  - [ ] I have modified/added tests to cover my changes.
- [ ] All existing tests pass.

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# PR Details
```
Comment (payneBrandon) [inline docs/pull_request_template.md:3]: Let's follow the ODE pr template a bit and add a few things:
- thanks for contributing
- section on how it was tested
- checkboxes for types of change (good for future contributions and tracking)

Also, we could probably hold off on this one a bit but down the line would be good - what about adding in a contributing guide and linking to it here as well?

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If a section applies, ensure you have met all of the associated checks -->
+
+- [ ] My changes require new environment variables:
+  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
+- [ ] My changes require updates to the documentation:
+  - [ ] I have updated the documentation accordingly.
+- [ ] My changes require updates and/or additions to the unit tests:
+  - [ ] I have modified/added tests to cover my changes.
+  - [ ] All existing tests pass.
```
Comment (payneBrandon) [inline docs/pull_request_template.md]: Let's pull this one out to the root level, even if they don't believe they have anything that would require unit test updates let's have them verify the existing ones pass.

**Review by payneBrandon — APPROVED** [review]
Just a couple comments, love the pr template idea!

File: docs/pull_request_template.md (hunk 1 above)
Comment (drewjj) [reply docs/pull_request_template.md:3]: Lol I thought the thanking part was goofy. I'll add it back

Comment (drewjj) [reply docs/pull_request_template.md:3]: Updated

File: docs/pull_request_template.md (hunk 2 above)
Comment (drewjj) [reply docs/pull_request_template.md]: Updated

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Review by Michael7371 — APPROVED** [review]
Looks good!

**Pull request merged**